frontend/src/components/MyApplicationItem/MyMissionItem/__tests__/useJudgment.test.ts

```diff
@@ -0,0 +1,155 @@
+    it("미션이 제출되지 않았을 때 isJudgmentAvailable이 false를 반환해야 한다", () => {
+      const unsubmittedMissionItem = { ...mockMissionItem, submitted: false };
+      const { result } = renderHook(() =>
+        useMissionJudgment({
+          missionItem: unsubmittedMissionItem,
+          recruitmentId: mockRecruitmentId,
+        })
+      );
+
+      expect(result.current.isJudgmentAvailable).toBe(false);
+    });
+
+    it("미션이 제출되었고, 예제 테스트 채점이 시작되지 않았을 때 isJudgmentAvailable이 true를 반환해야 한다", () => {
+      const missionWithoutJudgment = { ...mockMissionItem, judgment: null };
+      const { result } = renderHook(() =>
+        useMissionJudgment({
+          missionItem: missionWithoutJudgment,
+          recruitmentId: mockRecruitmentId,
+        })
+      );
+
+      expect(result.current.isJudgmentAvailable).toBe(true);
+    });
+
+    it("미션이 제출되었고, 예제 테스트 채점이 시작되었을 때 isJudgmentAvailable이 true를 반환해야 한다", () => {
+      (isJudgmentTimedOut as jest.Mock).mockReturnValue(true);
+      const { result } = renderHook(() =>
+        useMissionJudgment({ missionItem: mockMissionItem, recruitmentId: mockRecruitmentId })
+      );
+
+      expect(result.current.isJudgmentAvailable).toBe(true);
+    });
+
+    it("미션 상태가 제출 기간 이후 시점(ENDED)일 때 isJudgmentAvailable이 false를 반환해야 한다", () => {
+      const endedMission = { ...mockMissionItem, status: MISSION_STATUS.ENDED };
+      const { result } = renderHook(() =>
+        useMissionJudgment({ missionItem: endedMission, recruitmentId: mockRecruitmentId })
+      );
+
+      expect(result.current.isJudgmentAvailable).toBe(false);
+    });
+
+    it("미션 상태가 제출 기간 사이에 제출이 불가능한 경우(UNSUBMITTABLE)일 때 isJudgmentAvailable이 false를 반환해야 한다", () => {
+      const unsubmittableMission = { ...mockMissionItem, status: MISSION_STATUS.UNSUBMITTABLE };
+      const { result } = renderHook(() =>
+        useMissionJudgment({ missionItem: unsubmittableMission, recruitmentId: mockRecruitmentId })
+      );
+
+      expect(result.current.isJudgmentAvailable).toBe(false);
+    });
+  });
+
+  describe("fetchJudgmentMissionResult 테스트", () => {
+    it("판정 결과를 성공적으로 가져와야 한다", async () => {
+      const mockResponse = {
+        data: {
+          ...mockMissionItem.judgment,
+          status: JUDGMENT_STATUS.SUCCEEDED,
+          passCount: 8,
+          totalCount: 10,
+        },
+      };
+      (postMyMissionJudgment as jest.Mock).mockResolvedValue(mockResponse);
+
+      const { result } = renderHook(() =>
+        useMissionJudgment({ missionItem: mockMissionItem, recruitmentId: mockRecruitmentId })
+      );
+
+      await act(async () => {
+        const judgmentResult = await result.current.fetchJudgmentMissionResult();
+        expect(judgmentResult).toEqual({ ...mockMissionItem, judgment: mockResponse.data });
+        expect(judgmentResult.judgment?.status).toBe(JUDGMENT_STATUS.SUCCEEDED);
+        expect(judgmentResult.judgment?.passCount).toBe(8);
+        expect(judgmentResult.judgment?.totalCount).toBe(10);
+      });
```

Comment:
a: [expect.objectContaining(object)](https://jestjs.io/docs/expect#expectobjectcontainingobject)을 이용하면 좀 더 간결하게 테스트할 수 있을 것 같아요~!

// 예시
```suggestion
  await act(async () => {
    const judgmentResult = await result.current.fetchJudgmentMissionResult();
    expect(judgmentResult).toEqual(expect.objectContaining({
      judgment: expect.objectContaining({
        status: JUDGMENT_STATUS.SUCCEEDED,
        passCount: 8,
        totalCount: 10,
      }),
    }));
  });
```